independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -854,6 +861,14 @@ private Map<DotName, ClassInfo> findQualifiers() {
             if (isExcluded(qualifierClass)) {
                 continue;
             }
+            // check that all array typed methods are @Nonbinding
+            for (MethodInfo mi : qualifierClass.methods()) {
+                if (mi.returnType().kind().equals(Type.Kind.ARRAY) && mi.annotation(DotNames.NONBINDING) == null) {
+                    throw new DefinitionException("Qualifier annotation '" + qualifierClass + "' contains a method '"
```

Comment:
Would it make sense to resuse the logic from above?

Reply:
It would be only for the exception message which I considered but found awkward.
The detection logic is slightly  different as synthetic qualifiers don't inspect the annotation but instead the data provided by user.

Reply:
> which I considered but found awkward

Why? I would just add a private method like `DefinitionException newQualifierArrayValuedMemberException(ClassInfo qualifierClass, MethodInfo member)` to remove the duplicate error message...